publish_courses/static/publish_courses/js/controllers/IndexController.js

```diff
@@ -5,164 +5,88 @@
+                text: 'These courses will be published momentarily. ' +
+                'Please check the summary on this page, or the reports ' +
+                'in your account settings, to verify that all courses ' +
+                'are published.'},
+            failure: {
+                alertType: 'danger',
+                text: 'There was a problem publishing these courses.'} };
 
-        $scope.filterOptions = {
-            // `key` and `value` are the GET params sent to the server when
-            // the option is chosen. `value` must be unique in its option list,
-            // as it is also used for the HTML input element value. `value` can
-            // be arbitrarily set to something unique if there are duplicate
-            // values; in that case, an optional `query_value` attribute can be
-            // included in the option object to indicate what to send to the
-            // server. `query` is `true` if this option should trigger a GET
-            // param included in the request, or `false` if it is e.g. the
-            // default and should not be appended to the request params.
-            terms: [
-              // todo: move a 'no terms found' into the template, remove this
-                {key:'term', value: '', name:'', query: false, text: 'Choose a term <span class="caret"></span>'}
-                // specific terms are filled out dynamically below
-            ],
-        };
-        $scope.filters = {terms: $scope.filterOptions.terms[0]};
-
-        var schoolUrl = '/icommons_rest_api/api/course/v2/schools/'
-            + $scope.school.id + '/';
-        $http.get(schoolUrl)
-            .then(function successCallback(response) {
-                $scope.school.name = response.data.title_short || response.data.title_long;
-            }, $scope.handleAjaxErrorResponse);
+        // get school display name from the sis_account_id context
+        atrapi.Schools.get($scope.school.id)
+            .then(function gotSchool(r) {
+                $scope.school.name = r.title_short
+                                     || r.title_long
+                                     || $scope.school.id;});
 
-        // fetch active, complete terms for last five years; note this does not
-        // include Ongoing term (year==1900)
+        // fetch active, complete terms for last two years and any future years;
+        // note this does not include Ongoing term (year==1900)
         var currentYear = new Date().getFullYear();
-        var httpGetConfig = {params: {
-            active: 1,
-            calendar_year__gte: currentYear-4,  // last five years
-            limit: 100,
-            ordering: '-end_date,term_code__sort_order',
-            school: $scope.school.id,
-            with_end_date: 'True',
-            with_start_date: 'True',
-        }};
+        var termsGetConfig = { params: {
+                // calendar_year__gte: currentYear - 1,  // last two years
```

Comment:
Should we be getting terms from last 2 years? Maybe a question for PO. 
Also, a nitick: line 42 can be deleted.